Bottom: 24fdf03356e3f963a01b5230eefadcabb6deb956
Top:    bee4276839b9406d32b7160e1050216d2132af40
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 12:29:23 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index d5d08f1..feed3b6 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -306,6 +306,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
